core/src/main/java/smile/regression/treeshap/ShapCalculationTemp.java

```diff
@@ -0,0 +1,62 @@
+package smile.regression.treeshap;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import smile.util.MutableInt;
+
+/** 
+ * recyclable class as a register during shap calculation to avoid mass instantiation in memory 
+
+ * <pre>
+ * https://github.com/slundberg/shap/blob/master/shap/explainers/pytree.py
+ * </pre>
+ * 
+ * @author ray
+ */
+public class ShapCalculationTemp {
+
+  public int s;
+  public List<MutableInt> feature_indexes;
+  public List<MutableDouble> zero_fractions;
+  public List<MutableDouble> one_fractions;
+  public List<MutableDouble> pweights;
+
+  ShapCalculationTemp(int s) {
+    this.s = s;
+    this.feature_indexes = new ArrayList<MutableInt>(s);
+    this.zero_fractions = new ArrayList<MutableDouble>(s);
+    this.one_fractions = new ArrayList<MutableDouble>(s);
+    this.pweights = new ArrayList<MutableDouble>(s);
+
+    returnTemp(true);
+  }
+
+  void returnTemp(boolean init) {
```

Comment:
I don't feel that it is necessary to create this class if all we need is this function.